### PR TITLE
feat: add string filters

### PR DESCRIPTION
**Pull request by almeidabbm**
Closes: #11053 
Closes: #11056 

### Description:

image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/common/Filters/FilterInputs/FilterMultiStringInput.tsx [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
 import { Group, MultiSelect, Text, type MultiSelectProps } from '@mantine/core';
 import { IconPlus } from '@tabler/icons-react';
 import uniq from 'lodash/uniq';
 import { useCallback, useMemo, useState, type FC } from 'react';
 import MantineIcon from '../../MantineIcon';
-import { useFiltersContext } from '../FiltersProvider';
 import MultiValuePastePopover from './MultiValuePastePopover';
 
 type Props = Omit<MultiSelectProps, 'data' | 'onChange'> & {
-    field: FilterableItem;
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Filters/FilterInputs/FilterMultiStringInput.tsx:11]: is this component being used somewhere else?

File: packages/frontend/src/features/semanticViewer/components/BadgeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+type BadgeButtonProps = ButtonProps & {
+    onClick?: () => void;
+};
+
+const BadgeButton: FC<BadgeButtonProps> = ({ children, ...props }) => {
+    return (
+        <Button
+            variant="outline"
+            {...props}
+            // TODO: fix the stylings here
```
Comment (IrakliJani) [inline packages/frontend/src/features/semanticViewer/components/BadgeButton.tsx:13]: its fine as is for now

File: packages/frontend/src/features/semanticViewer/components/FiltersModal/FilterFieldSelectItem.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+const FilterFieldSelectItem = forwardRef<HTMLDivElement, ItemProps>(
+    ({ field, ...others }: ItemProps, ref) => {
+        return (
+            <div ref={ref} {...others}>
+                <Group noWrap>
+                    <FieldIcon field={field} />
+                    {field.label ?? field.name}
+                </Group>
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/features/semanticViewer/components/FiltersModal/FilterFieldSelectItem.tsx:17]: also, this is yet another field select component that we've implemented from scratch. we gotta unify those

File: packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    const handleApply = useCallback(() => {
+        onApply?.();
+        onClose?.();
+    }, [onApply, onClose]);
+
+    return (
+        <Modal
+            {...props}
+            onClose={onClose}
+            overlayProps={{ opacity: 0.1 }}
```
Comment (IrakliJani) [inline packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx:80]: hmm, isn't it too low?

File: packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+                        </ActionIcon>
+                    </Group>
+                )}
+                <Flex w="100%" justify="flex-end">
+                    <Button
+                        bg="black"
+                        onClick={handleApply}
+                        sx={(theme) => ({
+                            ':hover': {
+                                backgroundColor: theme.colors.gray[6],
```
Comment (IrakliJani) [inline packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx:171]: still unsure about this button. this is yet another custom styled button

File: packages/frontend/src/components/common/Filters/FilterInputs/FilterMultiStringInput.tsx (hunk 1 above)
Comment (almeidabbm) [reply packages/frontend/src/components/common/Filters/FilterInputs/FilterMultiStringInput.tsx:11]: Yes this component is being used for the inputs in the current explore, the field prop wasn't being used at all so I could just remove it and reuse the component

File: packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx (hunk 5 above)
Comment (almeidabbm) [reply packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx:171]: Yeah true, I'll keep the default button styling for now and we can change in the future if that's a problem

File: packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx (hunk 4 above)
Comment (almeidabbm) [reply packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx:80]: Might be, it looked like setting it higher would make it look too dark 🤔 

this is how it currently looks:
image

this is with 0.5

image

Comment (almeidabbm) [reply packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx:80]: perhaps set it to somewhere in the middle? like 0.3?

File: packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx (hunk 5 above)
Comment (IrakliJani) [reply packages/frontend/src/features/semanticViewer/components/FiltersModal/index.tsx:171]: yes please. as mentioned above, I am not excited about introducing new styles till we have a proper UI guide book

File: packages/frontend/src/features/sqlRunner/hooks/requestUtils.ts [hunk 6]
```diff
@@ -56,6 +56,9 @@ export const getResultsFromStream = async <T>(url: string | undefined) => {
         const jsonObjects: T[] = jsonStrings
             .map((jsonString) => {
                 try {
+                    if (!jsonString) {
+                        return null;
+                    }
```
[the remark this answers is not in the document]
Comment (almeidabbm) [reply packages/frontend/src/features/sqlRunner/hooks/requestUtils.ts:61]: Yes, the `getResultsFromStream` utility function is defined in `sqlRunner` and used in semantic viewer `packages/frontend/src/features/semanticViewer/api/requests.ts`

I just checked again the file and I am moving it to somewhere both features have in common, it only contains that function

[the remark this answers is not in the document]
Comment (almeidabbm) [reply packages/frontend/src/features/sqlRunner/hooks/requestUtils.ts:61]: Actually, both functions are reused across both features, moving the file to somewhere they have in common but leaving the function names and types intact (these will have to change in the future)

File: packages/frontend/src/utils/requestUtils.ts [hunk 7]
```diff
@@ -78,7 +81,7 @@ export const getResultsFromStream = async <T>(url: string | undefined) => {
     }
 };
 
-// recursively check job status until it's complete
+// TODO: this + type should be named something more generic since it is reused in multiple features
```
Comment (almeidabbm) [inline packages/frontend/src/utils/requestUtils.ts:84]: moved the file here and added a comment about the names @IrakliJani I don't think this should be done in this PR

**Pull request merged**